eras/shelley/impl/src/Cardano/Ledger/Shelley/Rewards.hs

```diff
@@ -158,6 +163,37 @@ aggregateRewards ::
 aggregateRewards pp rewards =
   Map.map (foldMap' rewardAmount) $ fst $ filterRewards pp rewards
 
+-- ================================================
+-- Compact Coin versions of sumRewards and aggregateCompactRewards
+
+sumCompactRewards ::
+  forall c pp.
+  (HasField "_protocolVersion" pp ProtVer) =>
+  pp ->
+  Map (Credential 'Staking c) (Set (Reward c)) ->
+  CompactForm Coin
+sumCompactRewards protocolVersion rs = fold $ aggregateCompactRewards protocolVersion rs
+
+-- | for each (Set (Reward c)) entry in the map, sum up the coin. In the ShelleyEra
+--   some of the coins are ignored (because of backward compatibility) see 'filterRewards'
+aggregateCompactRewards ::
+  forall c pp.
+  (HasField "_protocolVersion" pp ProtVer) =>
+  pp ->
+  Map (Credential 'Staking c) (Set (Reward c)) ->
+  Map (Credential 'Staking c) (CompactForm Coin)
+aggregateCompactRewards pp rewards =
+  Map.map (foldMap' (compactCoinOrError . rewardAmount)) $ fst $ filterRewards pp rewards
+
+-- We argue that the call to 'compactCoinOrError' will never return error.
+-- The Reward is is stored in the LedgerState, and we know the sum of all Ada in the LedgerState cannot
+-- exceed (maxBound :: Word64), So if the sum cannot excced it, neither can any component of the sum.
```

Comment:
Solid argument with regards to `compactCoinOrError` safety.
```suggestion
-- The Reward is stored in the LedgerState, and we know the sum of all Ada in the LedgerState cannot
-- exceed (maxBound :: Word64), So if the sum cannot exceed it, neither can any component of the sum.
```

Reply:
done